File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        pos=[]
        neg=[]
        for i in range(len(nums)):
            if(nums[i]>0):
                pos.append(nums[i])
            else:
                neg.append(nums[i])
        res=[]
        m=0
        k=0
        for i in range(len(nums)):
            if i%2==0:
                res.append(pos[m])
                m+=1
            else:
                res.append(neg[k])
                k+=1
        return res

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        List<Integer> pos=new ArrayList<>();
        List<Integer> neg=new ArrayList<>();
        List<Integer> res=new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            if(nums[i]>0){
                pos.add(nums[i]);
            }
            else{
                neg.add(nums[i]);
            }
        }
        int k=0,m=0;
        for(int i=0;i<nums.length;i++){
            if(i%2==0){
                res.add(pos.get(k));
                k++;
            }
            else{
                res.add(neg.get(m));
                m++;
            }
        }
        return res.stream().mapToInt(Integer::intValue).toArray();
    }
}

